GetF!t/Extensions/Date+.swift

```diff
@@ -12,7 +12,7 @@ extension Date {
     
     var yearMonthDay: String {
         let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "yyyy MM dd"
+        dateFormatter.dateFormat = DateProvider.yyyyMMdd
         return dateFormatter.string(from: self)
     }
```

Comment:
Toto je trochu blbé v tom, že zakaždým keď chceš vytvoriť `yearMontDay` tak nanovo vytváraš nový `DateFormatter`. Vytváranie DateFormatteru je expensive operation (to znamená, že to zaberá veľa času) a preto to chceš robiť iba keď musíš. [Tu si môžeš o tom pozrieť viac.](https://sarunw.com/posts/how-expensive-is-dateformatter/)

Navrhujem preto do tejto extension pridať aj statické date formátre:
```swift
static let yearMonthDayFormatter: DateFormatter = {
    let dateFormatter = DateFormatter()
    dateFormatter.dateFormat = DateProvider.yyyyMMdd
    return dateFormatter
}()

// a potom ho len reusovať namiesto toho aby si ho vždy inicializoval
var yearMonthDay: String {
       yearMonthDayFormatter.string(from: self)
}
```
Toto platí aj pre všetky ostatné computed properties čo máš v tejto extension.